controllers/rcj_soccer_referee_supervisor/referee/referee.py

```diff
@@ -67,18 +70,67 @@ def check_progress(self):
 
     def check_goal(self):
         """Check if goal is scored"""
+
+        team_goal = None
+        team_kickoff = None
+
         # ball in the blue goal
         if self.ball_translation[0] > GOAL_X_LIMIT:
             self.score_yellow += 1
-            self.draw_scores(self.score_blue, self.score_yellow)
-            self.ball_reset_timer = self.post_goal_wait_time
+
+            team_goal = self.team_name_yellow
+            team_kickoff = Team.BLUE
 
         # ball in the yellow goal
         elif self.ball_translation[0] < -GOAL_X_LIMIT:
             self.score_blue += 1
+
+            team_goal = self.team_name_blue
+            team_kickoff = Team.YELLOW
+
+        # If a goal was scored, redraw the scores, set the timers, log what
+        # happened and set the proper team for kickoff.
+        if team_goal and team_kickoff:
             self.draw_scores(self.score_blue, self.score_yellow)
             self.ball_reset_timer = self.post_goal_wait_time
 
+            self.log.event(
+                supervisor=self,
+                type=GameEvents.GOAL.value,
+                team=team_goal,
+                msg="A goal was scored by {team_goal}",
+                payload={
+                    "score_yellow": self.score_yellow,
+                    "score_blue": self.score_blue,
+                }
+            )
+
+            # Let the team that did not score the goal have a kickoff.
+            self.kickoff(team_kickoff)
+
+    def kickoff(self, team: Optional[Team] = None):
+        """Set up the kickoff by putting one of the robots of the team that is
+        kicking off closer to the center point
+
+        Args:
+            team (Team): The team that is kicking off. If the team is not
+                specified, it will be chosen randomly.
+        """
+        if team not in (Team.BLUE, Team.YELLOW, None):
+            raise ValueError(f'Unexpected team name {team}')
+
+        if not team:
+            team = Team.BLUE if random.random() > 0.5 else Team.YELLOW
```

Comment:
The cool thing here is that we should already know about it, thanks to the fact that we have a JSONL log from the first half.

What I had in mind for the second half was something like the following:
1. Figure out who kicked off the first in the first half
    ```bash
    $ cat The_Blues_vs_The_Yellows-2021-01-02T11:48:24.890235Z.jsonl | grep KICKOFF | head -n 1 | jq .
    "Y3"
    ```
2. Add an env variable (something like `RCJ_SOCCERSIM_KICKOFF_TEAM`) to `B` would be set in case yellow was kicking off in the first half and `Y` otherwise
3. Run the second half similarly to the first one

Reply:
Cool, sounds good to me.